convert2rhel/unit_tests/toolopts_test.py

```diff
@@ -145,3 +146,254 @@ def test_both_disable_submgr_and_no_rhsm_options_work(argv, raise_exception, no_
+        ),
+    ),
+)
+def test_config_file(argv, content, output, message, monkeypatch, tmp_path, caplog):
+    # After each test there were left data from previous
+    # Re-init needed delete the set data
+    convert2rhel.toolopts.tool_opts.__init__()
+    path = os.path.join(str(tmp_path), "convert2rhel.ini")
+    with open(path, "w") as file:  # pylint: disable=unspecified-encoding
+        file.write(content)
+    os.chmod(path, 0o600)
+
+    monkeypatch.setattr(sys, "argv", argv)
+    monkeypatch.setattr(convert2rhel.toolopts, "CONFIG_PATHS", value=[path])
+    convert2rhel.toolopts.CLI()
+
+    assert convert2rhel.toolopts.tool_opts.activation_key == output["activation_key"]
+    assert convert2rhel.toolopts.tool_opts.password == output["password"]
+    if message:
+        assert message in caplog.text
+
+
+@pytest.mark.parametrize(
+    ("argv", "content", "message", "output"),
+    (
+        (
+            mock_cli_arguments(["--password", "pass", "-f"]),
+            "pass_file",
+            "Password file argument take precedence over the password argument.",
+            {"password": "pass_file", "activation_key": None},
+        ),
+        (
+            mock_cli_arguments(["--password", "pass", "--config-file"]),
+            "[subscription_manager]\nactivation_key = key_cnf_file",
+            "Command line authentication method take precedence over method in configuration file.",
+            {"password": "pass", "activation_key": None},
+        ),
+    ),
+)
+def test_multiple_auth_src_combined(argv, content, message, output, caplog, monkeypatch, tmp_path):
+    """Test combination of password file or configuration file and CLI arguments."""
+    convert2rhel.toolopts.tool_opts.__init__()
+    path = os.path.join(str(tmp_path), "convert2rhel.file")
+    with open(path, "w") as file:  # pylint: disable=unspecified-encoding
+        file.write(content)
+    os.chmod(path, 0o600)
+    # The path for file is the last argument
+    argv.append(path)
+
+    monkeypatch.setattr(sys, "argv", argv)
+    monkeypatch.setattr(convert2rhel.toolopts, "CONFIG_PATHS", value=[""])
+    convert2rhel.toolopts.CLI()
+
+    assert message in caplog.text
+    assert convert2rhel.toolopts.tool_opts.activation_key == output["activation_key"]
+    assert convert2rhel.toolopts.tool_opts.password == output["password"]
+
+
+@pytest.mark.parametrize(
+    ("argv", "content", "message", "output"),
+    (
+        (
+            mock_cli_arguments(["--password", "pass", "-f", "file", "--config-file", "file"]),
+            ("pass_file", "[subscription_manager]\nactivation_key = pass_cnf_file"),
+            "Password file take precedence over the config file.",
+            {"password": "pass_file", "activation_key": None},
+        ),
+    ),
+)
+def test_multiple_auth_src_files(argv, content, message, output, caplog, monkeypatch, tmp_path):
+    """Test combination of password file, config file and CLI."""
```

Comment:
I would say we don't need to have docstrings in the test functions, in this case, it is pretty clear to me what the test is about.
```suggestion
```